src/test/java/de/medizininformatikinitiative/torch/FhirControllerIT.java

```diff
@@ -299,12 +299,29 @@ void emptyRequestBodyReturnsBadRequest() {
             assertThat(response.getBody()).contains("Empty request body");
         }
 
-        @ParameterizedTest
-        @ValueSource(strings = {"src/test/resources/CRTDL_Parameters/Parameters_invalid_CRTDL.json"})
-        void invalidCRTDLReturnsValidationException(String parametersFile) {
+        @Test
+        void invalidCRTDLReturnsBadRequest() {
+            TestRestTemplate restTemplate = new TestRestTemplate();
             HttpHeaders headers = new HttpHeaders();
             headers.add("content-type", "application/fhir+json");
-            testExecutor(parametersFile, "http://localhost:" + port + "/fhir/$extract-data", headers, 400);
+
+            try {
+                String fileContent = Files.readString(Paths.get("src/test/resources/CRTDL_Parameters/Parameters_invalid_CRTDL.json"), StandardCharsets.UTF_8);
+                HttpEntity<String> entity = new HttpEntity<>(fileContent, headers);
+                try {
+                    long start = System.nanoTime();
+                    ResponseEntity<String> response = restTemplate.exchange("http://localhost:" + port + "/fhir/$extract-data", HttpMethod.POST, entity, String.class);
+                    assertThat(response.getStatusCode().value()).isEqualTo(400);
+                    assertThat(durationSecondsSince(start)).isLessThan(1);
+                } catch (HttpStatusCodeException e) {
+                    logger.error("HTTP Status code error: {}", e.getStatusCode(), e);
+                    Assertions.fail("HTTP request failed with status code: " + e.getStatusCode());
+                }
+
+            } catch (IOException e) {
```

Comment:
please just throw exceptions from the test method. logging will not fail the test